tests/e2e/cucumber/steps/api.ts

```diff
@@ -126,3 +128,35 @@ Given(
     }
   }
 )
+
+Given(
+  '{string} creates the following file(s) in personal space using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    for (const info of stepTable.hashes()) {
+      await api.dav.uploadFileInPersonalSpace({
+        user,
+        pathToFile: info.pathToFile,
+        content: info.content
+      })
+    }
+  }
+)
+
+Given(
+  '{string} uploads following local file in personal space using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    const lastIndex = config.assets.lastIndexOf('/')
+    const requiredBasePathToLocalfile = config.assets.slice(0, lastIndex + 1)
+    for (const info of stepTable.hashes()) {
+      const pathToLocalFile = path.join(requiredBasePathToLocalfile, info.localFile)
```

Comment:
Can this be useful?
https://github.com/owncloud/web/blob/5626822b6004991fa1ad883723bf430a81512a9b/tests/e2e/support/environment/file.ts#L6-L16

Reply:
i will look into it  thanks !!